src/pages/index.js

```diff
@@ -0,0 +1,266 @@
+
+  cardNameElement.textContent = data.name;
+  cardImageElement.src = data.link;
+  cardImageElement.alt = data.name;
+
+  if (data.isLiked) {
+    cardLikeBtn.classList.add("card__like-button_liked");
+  }
+
+  cardLikeBtn.addEventListener("click", (evt) => {
+    handleCardLike(evt, data._id);
+  });
+
+  cardDeleteBtn.addEventListener("click", () =>
+    handleDeleteCard(cardElement, data._id)
+  );
+
+  cardImageElement.addEventListener("click", () => {
+    openModal(previewModal);
+    previewModalImage.src = data.link;
+    previewModalImage.alt = data.name;
+    previewModalPicDescription.textContent = data.name;
+  });
+  return cardElement;
+}
+function handleCardLike(evt, id) {
+  const isLiked = evt.target.classList.contains("card__like-button_liked");
+  api
+    .changeLikeStatus(id, isLiked)
+    .then(() => {
+      evt.target.classList.toggle("card__like-button_liked");
+    })
+    .catch(console.error);
+}
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", closeModalEscape);
+}
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", closeModalEscape);
+}
+previewModalCloseBtn.addEventListener("click", () => {
+  closeModal(previewModal);
+});
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, !true);
+    });
+}
+function handleCardFormSubmit(evt) {
+  evt.preventDefault();
+  const inputValues = {
+    name: cardModalNameInput.value,
+    link: cardModalLinkInput.value,
+  };
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+  api
+    .postInitialCards(inputValues)
+    .then((data) => {
+      previewModalPicDescription.textContent = data.name;
+      previewModalImage.src = data.link;
+      previewModalImage.alt = data.name;
+      const cardElement = getCardElement(data);
+      cardFormElement.reset();
+      cardsList.prepend(cardElement);
+      disableButton(cardSubmitBtn, settings);
+      closeModal(cardModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, !true);
+    });
+}
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+  api
+    .editAvatarInfo(avatarModalLinkInput.value)
+
+    .then((data) => {
+      profileAvatar.src = data.avatar;
+      profileAvatar.alt = data.avatar;
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, !true);
+    });
+}
```

Comment:
* I can't close the confirmation modal by clicking on its `close icon` or `cancel` https://snipboard.io/A4jlkg.jpg